### PR TITLE
rachel - L141.hfc R S T Y - take2

### PR DESCRIPTION
**Pull request by rhoesly**
This is a new PR from a clean master branch. All the comments from the last PR (rachel - L141.hfc R S T Y) are *hopefully* addressed, but there should be no merge conflicts/test fails. 

I am unsure if the code for silencing package check notes worked, and I couldn’t figure out the problem with the line endings.

I also tried updating the script that uses temp data injects of the outputs of this script. However, after changing to use outputs rather than temp files, the script (L241) no longer passes time shift tests so I did not include those changes in this PR.

#423

### CODE REVIEW

**Comment by bpbond** [comment]
I wish someone (perhaps me) hadn't added those stupid `.DS_Store` files.

**Comment by bpbond** [comment]
Thanks Rachel! Yeah L241 isn't your problem, but thanks for trying. I'll look.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 1]
```diff
@@ -51,67 +48,210 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+    F_gases <- list(EDGAR_HFC125, EDGAR_HFC134a, EDGAR_HFC143a,
+                    EDGAR_HFC152a, EDGAR_HFC227ea, EDGAR_HFC23,
+                    EDGAR_HFC236fa, EDGAR_HFC245fa, EDGAR_HFC32,
+                    EDGAR_HFC365mfc, EDGAR_HFC43)
+
+    # Use function to convert datat to numeric values and combine to one tibble
+    # Map to GCAM countries and technologies
+    F_gases_formatted <- lapply(F_gases, format_F_Gas)
+
+    L141.EDGAR_HFC <- do.call(bind_rows,F_gases_formatted) %>%
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: FYI this is unnecessary: `bind_rows` accepts a list of data frames along with explicitly-provided objects. I'll make this change.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 2]
```diff
@@ -51,67 +48,210 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate(adj_emissions = emissions * scaler) %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year) %>%
+      summarise(adj_emissions = sum(adj_emissions, na.rm = T)) %>%
+      replace_na(list(adj_emissions = 0)) %>%  #replace NAs with zero. Some sectors have zero emissions.
+      rename(value = adj_emissions)
+
+    # Compute final cooling HFC emissions factors
+    L141.hfc_ef_R_cooling_Yh <- L141.hfc_R_S_T_Yh %>%
+      filter(supplysector %in% c("comm cooling", "resid cooling"), year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L141.R_cooling_T_Yh.long %>% select(GCAM_region_ID, year,service,value) %>% rename(energy = value),
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: FYI could just do `select(GCAM_region_ID, year, service, energy = value)`. I'll make this change.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 3]
```diff
@@ -51,67 +48,210 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Silence Package check notes
+    "." <- "year" <- "value" <- "GCAM_region_ID" <- "sector" <- "fuel" <- "service" <-
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: In the future, these shouldn't be in quotes, and need to assign them all to `NULL`. I'll change.

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 4]
```diff
@@ -51,67 +48,210 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+      rename(SSP2_tot = Emissions)
+
+    # Add SSP2 values to EDGAR HFC emissions then calculate scalar
+    L141.hfc_scaler <- L141.hfc_R_S_T_Yh_share %>% group_by(year,Non.CO2) %>%
+      summarize(EDGAR_tot = sum(emissions))%>%
+      arrange(Non.CO2,year) %>%
+      left_join(HFC_Inventory_GV, by = c(year = "Year", Non.CO2 = "Species")) %>% # some entries not in GUUS data, default to scaler =1
+      mutate(scaler = SSP2_tot / EDGAR_tot) %>%
+      replace_na(list(scaler = 1)) %>%
+      mutate_at(vars(), funs(replace(., is.infinite(.), 1))) %>% # replace na and Infinite values with 1.
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: Why? This step isn't in the original code that I see (lines 121-123 in `L141.hfc_R_S_T_Y.R`).

File: inst/extdata/emissions/EDGAR/EDGAR_HFC125.csv [hunk 5]
```diff
@@ -0,0 +1,101 @@
+# File: EDGAR_HFC125.csv
+# Title: EDGAR F fases trend
```
Comment (cahartin) [inline inst/extdata/emissions/EDGAR/EDGAR_HFC125.csv]: fgases

File: R/zchunk_L141.hfc_R_S_T_Y.R [hunk 6]
```diff
@@ -51,68 +47,211 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
...
+
+    # Add SSP2 values to EDGAR HFC emissions then calculate scalar
+    L141.hfc_scaler <- L141.hfc_R_S_T_Yh_share %>%
+      group_by(year, Non.CO2) %>%
+      summarize(EDGAR_tot = sum(emissions))%>%
+      arrange(Non.CO2, year) %>%
+      left_join(HFC_Inventory_GV, by = c(year = "Year", Non.CO2 = "Species")) %>% # some entries not in GUUS data, default to scaler =1
+      mutate(scaler = SSP2_tot / EDGAR_tot) %>%
+      replace_na(list(scaler = 1)) %>%
+      mutate_at(vars(), funs(replace(., is.infinite(.), 1))) %>% # replace na and Infinite values with 1.
```
Comment (bpbond) [inline R/zchunk_L141.hfc_R_S_T_Y.R]: I don't want previous comment to get lost--why are we doing this? It's not in original code that I see (and very confusing line here).

**Comment by bpbond** [comment]
Hi @rhoesly - I've done the downstream chunk stitching so this is all ready to go except Corinne's spelling catch and my query about line 181 above.

**Comment by bpbond** [comment]
Hey @rhoesly - reminder that this is waiting on a couple very small things I think.

**Comment by bpbond** [comment]
Hey @rhoesly - do you have time to finish this up?

**Comment by bpbond** [comment]
This will generate an error because of #677 but merging.

**Pull request merged**